src/Acuminator/Acuminator.Utilities/Roslyn/NestedInvocationWalker.cs

```diff
@@ -58,14 +59,40 @@ public abstract class NestedInvocationWalker : CSharpSyntaxWalker
 
 		private readonly Stack<SyntaxNode> _nodesStack = new Stack<SyntaxNode>();
         private readonly HashSet<IMethodSymbol> _methodsInStack = new HashSet<IMethodSymbol>();
+		private readonly Func<IMethodSymbol, bool> _bypassMethod;
 
-        protected NestedInvocationWalker(Compilation compilation, CancellationToken cancellationToken)
+		/// <summary>
+		/// Constructor of the class
+		/// </summary>
+		/// <param name="compilation">Compilation</param>
+		/// <param name="cancellationToken">Cancellation token</param>
+		/// <param name="bypassMethod">Delegate to control if it is needed to bypass analysis of an invocation of a method and do not step into it. If not supplied, default implementation is used to bypass some core types from PX.Data namespace</param>
+		protected NestedInvocationWalker(Compilation compilation, CancellationToken cancellationToken,
+			Func<IMethodSymbol, bool> bypassMethod = null)
 		{
 			compilation.ThrowOnNull(nameof (compilation));
 
 			_compilation = compilation;
             CancellationToken = cancellationToken;
 
+			if (bypassMethod != null)
+			{
+				_bypassMethod = bypassMethod;
+			}
+			else
+			{
+				var pxContext = new PXContext(_compilation);
+				var typesToBypass = new[]
```

Comment:
@metallmonster maybe it makes sense to refactor out creation of forbidden types register - put it into separate field or generate in a virtual method. This way the inheritors of nested walker will have the ability to extend the set of types to which they should not descend

Reply:
Ok